Containerfile.base

```diff
@@ -1,4 +1,4 @@
-ARG RUBY_VERSION=3.0.6
+ARG RUBY_VERSION=3.1.4
```

Comment:
This doesn't actually need to be changed, it gets pulled from `.ruby-version` automatically as part of [`scripts/build_base_ruby_image.sh`](/scripts/build_base_ruby_image.sh).

That said, this bump also doesn't hurt anything. Maybe I should add a comment here for clarity for next time...